server/mergin/sync/models.py

```diff
@@ -833,7 +834,71 @@ def files(self) -> List[ProjectFile]:
             WHERE fh.change != 'delete';
         """
         params = {"project_id": self.project_id, "version": self.name}
-        result = db.session.execute(query, params).fetchall()
+        return db.session.execute(query, params).fetchall()
+
+    def _files_backward_search(self):
+        """Calculate version files using lookup from the last version"""
+        query = f"""
+            WITH files_changes_before_version AS (
+                WITH files_candidates AS (
+                    -- files removed later (but created anytime)
+                    SELECT DISTINCT
```

Comment:
Did you want the DISTINCT to apply to the union of the two selects or just this first one? By the way UNION (rather than UNION ALL) already ensures that the union is not adding duplicates to the first query